PhysicsTools/PatAlgos/plugins/PATIsolatedTrackProducer.cc

```diff
@@ -374,6 +382,16 @@ void pat::PATIsolatedTrackProducer::produce(edm::Event& iEvent, const edm::Event
 
     int trackQuality = gentk.qualityMask();
 
+    // get combined probQ and probXY
+    float probQonTrack = 0, probXYonTrack = 0, probQonTrackNoLayer1 = 0, probXYonTrackNoLayer1 = 0;
+    if (gt2siPixelTrackProbQXY.isValid() && gt2siPixelTrackProbQXY->contains(tkref.id())) {
```

Comment:
I have a preference to avoid `smart` checks for present data to avoid misconfigured jobs.

Admittedly, I let it in during review #19025, and this code apparently just follows the existing coding pattern.
The miniAOD configuration control has improved since 2017. So, it's better to rely on configuration with this update.

Please drop the check for `gt2siPixelTrackProbQXY.isValid()` here and replace with a test for `gt2siPixelTrackProbQXY_.isUninitialized()` or a bool computed during configuration